mscr-ui/src/modules/form/crosswalk-form/target-and-source-schema-selector.tsx

```diff
@@ -56,39 +85,113 @@ export default function TargetAndSourceSchemaSelector({
+      setDefaultSourceSchema(
+        getLanguageVersion({
+          data: sourceSchemaData.label,
+          lang,
+        })
+      );
+      setDefaultTargetSchema(
+        getLanguageVersion({
+          data: targetSchemaData.label,
+          lang,
+        })
+      );
     }
   }, [lang, schemaSelectorDisabled, sourceSchemaData, targetSchemaData]);
 
+  const workspaceValuesPersonalCrosswalks: SelectableWorkspace[] = [
+    {
+      labelText: 'All',
+      uniqueItemId: 'all',
+    },
+    {
+      labelText: 'Personal workspace',
+      uniqueItemId: 'personalWorkspace',
+    },
+  ];
+
+  const workspaceValuesGroupCrosswalks: SelectableWorkspace[] = [
+    {
+      labelText: 'All',
+      uniqueItemId: 'all',
+    },
+    {
+      labelText: 'Group workspace',
+      uniqueItemId: 'groupWorkspace',
+    },
+  ];
+
   useEffect(() => {
-    const fetchedSchemas: { labelText: string; uniqueItemId: string }[] = [];
+    const fetchedSchemas: SelectableSchema[] = [];
     data?.hits.hits.forEach((item: MscrSearchResult) => {
       const label = getLanguageVersion({
         data: item._source.label,
         lang,
       });
-      const schema = { labelText: label, uniqueItemId: item._source.id };
+
+      const schema = {
+        labelText: label,
+        uniqueItemId: item._source.id,
+        organization:
+          item._source.organizations.length > 0
+            ? item._source.organizations[0].id
+            : '',
+        owner:
+          item._source?.owner && item._source?.owner?.length > 0
+            ? item._source.owner
+            : [],
+      };
       fetchedSchemas.push(schema);
     });
     setDefaultSchemas(fetchedSchemas);
+    setSourceSchemas(fetchedSchemas);
+    setTargetSchemas(fetchedSchemas);
     setDataLoaded(true);
-    // console.log('schemas', defaultSchemas);
+    setSelectedSourceWorkspace('all');
+    setSelectedTargetWorkspace('all');
+
+    if (router.asPath.includes('personal')) {
+      setWorkspaceValues([...workspaceValuesPersonalCrosswalks]);
+    } else {
+      setWorkspaceValues([...workspaceValuesGroupCrosswalks]);
+    }
   }, [data?.hits.hits, isSuccess, lang]);
 
+  useEffect(() => {
+    if (selectedSourceWorkspace === 'all') {
+      setSourceSchemas(defaultSchemas);
+    } else if (selectedSourceWorkspace === 'personalWorkspace') {
+      setSourceSchemas(
+        defaultSchemas.filter((item) => item.owner.includes(user.id))
+      );
+    } else {
+      setSourceSchemas(
+        defaultSchemas.filter((item) => item.organization !== '')
+      );
+    }
+  }, [selectedSourceWorkspace]);
+
+  useEffect(() => {
+    if (selectedTargetWorkspace === 'all') {
+      setTargetSchemas(defaultSchemas);
+    } else if (selectedTargetWorkspace === 'personalWorkspace') {
+      setTargetSchemas(
+        defaultSchemas.filter((item) => item.owner.includes(user.id))
+      );
+    } else {
+      setTargetSchemas(
+        defaultSchemas.filter((item) => item.organization !== '')
```

Comment:
Shouldn't this be done like the owner is, so you get all organizations in the item.organization, and then check if it includes the organization whose workspace the modal was opened from? The group pid is already passed to the modals, so you just need to thread it through CrosswalkFormFields to get it here as well.

Although as discussed, eventually we'll want to implement this so that the selected filter determines what to fetch from the backend, so I don't know how much more effort we want to put to the interim solution.